Azure/azure-cli
# {RDBMS} `az postgres flexible-server migration create`: Update migration create help text for postgres flex

**Related command**
az postgres flexible-server migration create

**Description**
[az postgres flexible-server migration | Microsoft Learn](https://learn.microsoft.com/en-us/cli/azure/postgres/flexible-server/migration?view=azure-cli-latest) ---> During the create command, not able to see what the parameters are relevant in the JSON properties file.
This will help users in going over the CLI documentation and understanding the different parameters supported in the JSON.
You can refer this link - [How to set up Azure CLI for migration service in Azure Database for PostgreSQL - Flexible Server - Azure Database for PostgreSQL - Flexible Server | Microsoft Learn](https://learn.microsoft.com/en-us/azure/postgresql/migrate/migration-service/how-to-setup-azure-cli-commands-postgresql)

**Testing Guide**
Manual
```
Examples
    Start a migration workflow on the target server identified by the parameters. The configurations
    of the migration should be specified in the properties file. The different properties are
    defined as:
    sourceDbServerResourceId: Source server details.
    adminCredentials: This parameter lists passwords for admin users for both the source server and
    the target PostgreSQL flexible server.
    targetServerUserName: The default value is the admin user created during the creation of the
    PostgreSQL target flexible server, and the password provided is used for authentication against
    this user.
    dbsToMigrate: Specify the list of databases that you want to migrate to Flexible Server.
    overwriteDBsInTarget: When set to true (default), if the target server happens to have an
    existing database with the same name as the one you're trying to migrate, the migration service
    automatically overwrites the database.
    Sample migrationConfig.json for PostgreSQLSingleServer shown below.
    {
      "properties": {
        "sourceDBServerResourceId": "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-
    xxxxxxxxxxxx/resourceGroups/test-single-rg/providers/Microsoft.DBforPostgreSQL/servers/pg-
    single-1",
        "secretParameters": {
          "adminCredentials": {
            "sourceServerPassword": "password",
            "targetServerPassword": "password"
          },
          "sourceServerUserName": "testuser@pg-single-1",
          "targetServerUserName": "fspguser"
        },
        "dBsToMigrate": [
          "postgres"
        ],
        "overwriteDbsInTarget": "true"
      }
    }

        az postgres flexible-server migration create --subscription xxxxxxxx-xxxx-xxxx-xxxx-
        xxxxxxxxxxxx --resource-group testGroup --name testserver       --migration-name
        testmigration --properties "migrationConfig.json"


    Start a migration workflow on the target server identified by the parameters. The configurations
    of the migration should be specified in the migrationConfig.json file.       Use --migration-
    mode offline for Offline migration.

        az postgres flexible-server migration create --subscription xxxxxxxx-xxxx-xxxx-xxxx-
        xxxxxxxxxxxx --resource-group testGroup --name testserver       --migration-name
        testmigration --properties "migrationConfig.json" --migration-mode offline


    Start a migration workflow on the target server identified by the parameters. The configurations
    of the migration should be specified in the migrationConfig.json file.       Use --migration-
    mode online for Online(with CDC) migration. Use migration-option Validate for validate only
    request.

        az postgres flexible-server migration create --subscription xxxxxxxx-xxxx-xxxx-xxxx-
        xxxxxxxxxxxx --resource-group testGroup --name testserver       --migration-name
        testmigration --properties "migrationConfig.json" --migration-mode online --migration-option
        Validate


    Start a migration workflow on the target server identified by the parameters. The configurations
    of the migration should be specified in the migrationConfig.json file.       Use --migration-
    option Migrate for Migrate Only request.

        az postgres flexible-server migration create --subscription xxxxxxxx-xxxx-xxxx-xxxx-
        xxxxxxxxxxxx --resource-group testGroup --name testserver       --migration-name
        testmigration --properties "migrationConfig.json" --migration-option Migrate


    To start a migration for other than PostgreSQLSingleServer, soureType and sslMode must be
    specified in properties file. These properties are defined as:
    sourceType: Values can be - on-premises, AWS_RDS, AzureVM, PostgreSQLSingleServer
    sslMode:  SSL modes for migration. SSL mode for PostgreSQLSingleServer is VerifyFull and
    Prefer/Require for other source types.
    Sample migrationConfig.json shown below.
    {
      "properties": {
        "sourceDBServerResourceId": "<<hostname or IP address>>:<<port>>@<<username>>",
        "secretParameters": {
          "adminCredentials": {
            "sourceServerPassword": "password",
            "targetServerPassword": "password"
          },
          "sourceServerUserName": "postgres",
          "targetServerUserName": "fspguser"
        },
        "dBsToMigrate": [
          "ticketdb","timedb","inventorydb"
        ],
        "overwriteDbsInTarget": "true",
        "sourceType": "OnPremises",
        "sslMode": "Prefer"
      }
    }

        az postgres flexible-server migration create --subscription xxxxxxxx-xxxx-xxxx-xxxx-
        xxxxxxxxxxxx --resource-group testGroup --name testserver         --migration-name
        testmigration --properties "migrationConfig.json"


    Start a private endpoint enabled migration workflow on the target server by specifying
    migrationRuntimeResourceId in properties file. This property is defined as:
    migrationRuntimeResourceId: The resource ID of the migration runtime server that is responsible
    for migrating data between source and target server.
    Sample migrationConfig.json shown below.
    {
      "properties": {
        "sourceDBServerResourceId": "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-
    xxxxxxxxxxxx/resourceGroups/test-single-rg/providers/Microsoft.DBforPostgreSQL/servers/pg-
    single-1",
        "migrationRuntimeResourceId": "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceG
    roups/testGroup/providers/Microsoft.DBforPostgreSQL/flexibleServers/testsourcemigration",
        "secretParameters": {
          "adminCredentials": {
            "sourceServerPassword": "password",
            "targetServerPassword": "password"
          },
          "sourceServerUserName": "testuser@pg-single-1",
          "targetServerUserName": "fspguser"
        },
        "dBsToMigrate": [
          "postgres"
        ],
        "overwriteDbsInTarget": "true"
      }
    }

        az postgres flexible-server migration create --subscription xxxxxxxx-xxxx-xxxx-xxxx-
        xxxxxxxxxxxx --resource-group testGroup --name testserver --migration-name testmigration
        --properties "migrationConfig.json"
```


---

This checklist is used to make sure that common guidelines for a pull request are followed.
